new-lamassu-admin/src/pages/UserManagement/modals/ChangeRoleModal.js

```diff
@@ -0,0 +1,93 @@
+import { useMutation } from '@apollo/react-hooks'
+import { makeStyles } from '@material-ui/core/styles'
+import gql from 'graphql-tag'
+import React, { useState } from 'react'
+
+import Modal from 'src/components/Modal'
+import { Button } from 'src/components/buttons'
+import { Info2, P } from 'src/components/typography'
+
+import styles from '../UserManagement.styles'
+
+import Input2FAModal from './Input2FAModal'
+
+const CHANGE_USER_ROLE = gql`
+  mutation changeUserRole(
+    $confirmationCode: String
+    $id: ID!
+    $newRole: String!
+  ) {
+    changeUserRole(
+      confirmationCode: $confirmationCode
+      id: $id
+      newRole: $newRole
+    ) {
+      id
+    }
+  }
+`
+
+const useStyles = makeStyles(styles)
+
+const ChangeRoleModal = ({ state, dispatch, user, requiresConfirmation }) => {
+  const classes = useStyles()
+
+  const [changeUserRole] = useMutation(CHANGE_USER_ROLE, {
+    refetchQueries: () => ['users']
+  })
+
+  const [confirmation, setConfirmation] = useState(null)
+
+  const submit = () => {
+    changeUserRole({
+      variables: {
+        confirmationCode: confirmation,
+        id: user.id,
+        newRole: user.role === 'superuser' ? 'user' : 'superuser'
+      }
+    })
+    handleClose()
```

Comment:
Ok, didn't check every modal but I have a feeling it's happening everywhere, basically you wanna wait before the success response from the server before closing the modal. You also wanna show-up a error somewhere it if happen